### ISSUE TITLE
error initializing builder 'upcloud': Incompatible API version with plugin.

### ISSUE THREAD

**Issue by bbros-dev** [UpCloudLtd/upcloud-packer#22]
Using UpCloud builder version `3.0.0`:

```bash
2020/11/10 19:13:23 [INFO] Packer version: 1.6.4 [go1.15.2 linux amd64]
2020/11/10 19:13:23 [DEBUG] Discovered plugin: upcloud = /home/<redacted>/.packer.d/plugins/packer-builder-upcloud
2020/11/10 19:13:23 [DEBUG] Discovered plugin: vultr = /home/<redacted>/.packer.d/plugins/packer-builder-vultr
2020/11/10 19:13:23 using external builders [upcloud vultr]
2020/11/10 19:13:23 Checking 'PACKER_CONFIG' for a config file path
2020/11/10 19:13:23 'PACKER_CONFIG' not set; checking the default config file path
2020/11/10 19:13:23 Attempting to open config file: /home/<redacted>/.packerconfig
2020/11/10 19:13:23 [WARN] Config file doesn't exist: /home/<redacted>/.packerconfig
2020/11/10 19:13:23 Setting cache directory: /home/<redacted>/src/kfchub.tf/packer/packer_cache
2020/11/10 19:13:23 Creating plugin client for path: /home/<redacted>/.packer.d/plugins/packer-builder-upcloud
2020/11/10 19:13:23 Starting plugin: /home/<redacted>/.packer.d/plugins/packer-builder-upcloud []string{"/home/<redacted>/.packer.d/plugins/packer-builder-upcloud"}
2020/11/10 19:13:23 Waiting for RPC address for: /home/<redacted>/.packer.d/plugins/packer-builder-upcloud
2020/11/10 19:13:23 packer-builder-upcloud plugin: 2020/11/10 19:13:23 Plugin minimum port: 10000
2020/11/10 19:13:23 packer-builder-upcloud plugin: 2020/11/10 19:13:23 Plugin maximum port: 25000
2020/11/10 19:13:23 packer-builder-upcloud plugin: 2020/11/10 19:13:23 Plugin address: unix /tmp/packer-plugin340163571
2020/11/10 19:13:23 packer-builder-upcloud plugin: 2020/11/10 19:13:23 Waiting for connection...

error initializing builder 'upcloud': Incompatible API version with plugin.
Plugin version: 4, Ours: 5

2020/11/10 19:13:23 Build debug mode: true
2020/11/10 19:13:23 Force build: false
2020/11/10 19:13:23 On error: 
2020/11/10 19:13:23 Waiting on builds to complete...
==> Wait completed after 5 microseconds
Error: Failed to initialize build "upcloud"
==> Builds finished but no artifacts were created.

error initializing builder 'upcloud': Incompatible API version with plugin.
Plugin version: 4, Ours: 5


Debug mode enabled. Builds will not be parallelized.

==> Wait completed after 5 microseconds

==> Builds finished but no artifacts were created.
2020/11/10 19:13:23 [INFO] (telemetry) Finalizing.
2020/11/10 19:13:23 /home/<redacted>/.packer.d/plugins/packer-builder-upcloud: plugin process exited
2020/11/10 19:13:24 waiting for all plugin processes to complete...

```

**Comment by rubiesonthesky** [comment]
Try to use version 1.1.3 of packer as that is the version they are using in the tutorial. https://upcloud.com/community/tutorials/upcloud-packer-builder/  For mac users, that version is not available from brew, you need to manually download and unpack it.

**Comment by maxfrei** [comment]
The issue solved in release 4.0.0.
 https://github.com/UpCloudLtd/upcloud-packer/releases/tag/4.0.0
Closing.
